### DISCUSSION TITLE
How to always show workspaces specific to each monitor?

### DISCUSSION THREAD

**Discussion by nktnet1** [JakeStanger/ironbar#749] [Q&A]
I have 3 monitors, and I want the following:
- **Monitor 1**: show workspaces 1-10
- **Monitor 2**: show workspaces 2-20
- **Monitor 3**: show workspaces 3-30

I've tried using the [workspaces configuration](https://github.com/JakeStanger/ironbar/wiki/workspaces#configuration) `favorites = ["1", "2", ..., "30"]` with `all_monitors = false`, but then 1-30 shows up on all monitors.

---

For the full context, I want to replicate this Waybar's [hyprland/workspaces](https://github.com/Alexays/Waybar/wiki/Module:-Hyprland#workspaces) configuration:
```jsonc
    "hyprland/workspaces": {
        "persistent-workspaces": {
             "*": 10
        }
    },
```
while using the [hyprsplit](https://github.com/shezdy/hyprsplit) extension.

Ultimately, the goal is to make it look similar to [DWM](https://dwm.suckless.org/)'s default way of displaying tags (workspaces), where each monitor shows its own 10 available tags.

**Comment by JakeStanger** [comment]
To achieve this, you'd need a different configuration for the bar on each monitor. Ironbar does support this like so:

```corn
{
  monitors.DP-1.start = [ { type = "workspaces" favorites = [ "1" "2" "3" ... ] }
  monitors.DP-2.start = [ { type = "workspaces" favorites = [ "11" "12" "13" ... ] }
  monitors.DP-1.start = [ { type = "workspaces" favorites = [ "21" "22" "32" ... ] }
}
```

(I'd definitely recommend corn for this as you can use a `let` block to re-use the configs for other modules)

**Reply by nktnet1** [reply]
Thank you for replying @JakeStanger - the monitor port names changes for me arbitrarily so I'm not sure if this is a viable approach.

For example, sometimes it's DP-1/DP-2, other times it's DP-4/DP-8. Sometimes they also swap.

In Hyprland config, I was able to work around this by using the monitor's description itself, e.g. something like:
```
monitor = desc:BOE 0x0A2C,2560x1600@165, auto, 1.6
monitor = desc:Hewlett Packard HP LA2206 CNC1320BJL, 1920x1080@60, auto-left, 1
monitor = desc:Hewlett Packard HP LA2206 CNC1190S03, 1920x1080@60, auto-right, 1
```
and then in Waybar, use the `*` syntax (to acccount for the swap, i.e. not hard-coding workspaces to specific monitors)
```
    "hyprland/workspaces": {
        "persistent-workspaces": {
             "*": 10
        }
    },
```

although I'm not sure if this is possible in Ironbar.

I couldn't find any mention of it in https://github.com/JakeStanger/ironbar/wiki/configuration-guide unfortunately.

**Reply by JakeStanger** [reply]
Odd, those should be stable. AMD/NVidia/Intel GPU out of interest?

Currently Ironbar only supports using the name. Adding support for using the description isn't out of the question, but I won't be adding it until the next GTK3 release due to #642.

**Reply by nktnet1** [reply]
I'm using a hybrid GPU laptop with Intel and Nvidia (nouveau drivers). The monitors are connected to a docking station that uses Thunderbolt 4 (although the same issue can be reproduced on my DisplayLink docking station, with Nvidia disabled to strictly use intel).

I've seen quite a few instances of this happening online though so not sure if it's unique to my case - for example, here's someone (on Xorg too, not just Wayland):
- https://askubuntu.com/questions/1477245/external-monitors-sometimes-switch-names-when-docking-station-plugged-in/1518469?noredirect=1#comment2670566_1518469
